posts/2021-03-24-microprofile40-open-liberty-21003.adoc

```diff
@@ -0,0 +1,442 @@
+
+Also, make sure to configure your Liberty server with the `microProfile-4.0` feature in the server.xml:
+[source,xml]
+----
+<server>
+  <featureManager>
+    <feature>microProfile-4.0</feature>
+    <!-- ... -->
+  </featureManager>
+  <!-- ... -->
+</server>
+----
+
+
+That's it! Now that we've got our development and deployment environments set up, it's time to dive into each individual MicroProfile Specification to learn the amazing updates!
+
+. <<config-2.0, What's new in MicroProfile Config 2.0>>
+. <<fault-tolerance-3.0, New features in MicroProfile Fault Tolerance 3.0>>
+. <<health-3.0, New changes in MicroProfile Health 3.0>>
+. <<metrics-3.0, What's New in MicroProfile Metrics 3.0>>
+. <<open-tracing-2.0, New updates in  MicroProfile Open Tracing 2.0>>
+. <<open-api-2.0, What's New in MicroProfile Open API 2.0>>
+. <<rest-client-2.0, Awesome features in MicroProfile Rest Client 2.0>>
+. <<jwt-1.2, Must known features in MicroProfile JWT Auth 1.2>>
+
+
+
+[#config-2.0]
+== What's New in MicroProfile Config 2.0
+
+Quite a few nice capabilities were provided by MicroProfile Config 2.0, such as Config Profile, Config Properties, ConfigValue support, and more.
+
+Open Liberty's MicroProfile Config 2.0 feature is based on link:https://github.com/smallrye/smallrye-config[SmallRye's implementation] of the specification.
+
+Watch this space for a new blog post dedicated to MicroProfile Config 2.0.
+
+[#fault-tolerance-3.0]
+== What's New in MicroProfile Fault Tolerance 3.0
+
+MicroProfile Fault Tolerance allows you to easily apply strategies for mitigating failure to their code. It provides annotations which you can add to methods to use bulkhead, circuit breaker, retry, timeout and fallback strategies.
+
+Fault Tolerance 3.0 introduces two main features.
+
+=== MicroProfile Metrics tags
+
+MicroProfile Fault Tolerance 3.0 overhauls the metrics that are automatically exported, to take advantage of tags. Tags make it easier to use those metrics, to understand when and where your application is failing.
+
+Information which was previously included in the metric name is now included as metric tags. This makes it much easier to query for data from multiple methods and pick out those which may be causing an issue.
+
+Example:
+
+Old metric: `application:ft.<name>.timeout.callsTimedOut.total`
+
+New metric: `base:ft.timeout.calls.total{method="<name>", timedOut="true"}``
+
+
+=== Clarify the scope of Bulkheads and CircuitBreakers
+
+The specification now requires that circuit breakers and bulkheads are singletons, identified by the bean class and the guarded method. For example, if a @RequestScoped bean has a @CircuitBreaker method, all invocations of that method will share the same circuit breaker state, even though each request has a different instance of the bean.
```

Comment:
Should we not make "@RequestScoped" and "@CircuitBreakerr" inline code blocks? e.g: @RequestScoped and @CircuitBreaker @tevans78  @GraceJansen 

```suggestion
The specification now requires that circuit breakers and bulkheads are singletons, identified by the bean class and the guarded method. For example, if a `@RequestScoped` bean has a `@CircuitBreaker` method, all invocations of that method will share the same circuit breaker state, even though each request has a different instance of the bean.
```